### ISSUE TITLE
logs fill up disk

### ISSUE THREAD

**Issue by anarcat** [ipfs/go-ipfs#866]
so I imported a 3.2GB file into ipfs using `ipfs add`. apparently, that's an exotic enough operation to cause all sorts of problems. :) first it duplicated the file to `~/.go-ipfs` (to be expected?) and took around an hour (i was surprised by that) - but I guess that's a separate issue.

then i tried to share the file with a friend. it didn't work so well, things stalled and wouldn't work. so i went around to try to `ipfs cat > /dev/null` the file locally. what happened next was that my HDD was filled up with the daemon's logfiles. removing the files of course didn't help until i restarted the daemon, and then i was able to complete the `ipfs cat`.

**Comment by jbenet** [comment]
> first it duplicated the file to ~/.go-ipfs (to be expected?) 

Yep.

> took around an hour (i was surprised by that) - but I guess that's a separate issue.

that's abysmal perf-- something very weird is going on. what are the resources you're running this under? 
are you behind a NAT? was your friend?

> filled up with the daemon's logfiles

We chatted on irc and your log files do look like this:

```
> ipfs config Log
{
  "MaxAgeDays": 0,
  "MaxBackups": 1,
  "MaxSizeMB": 250
}
```
Which means that the log-rotator  `ipfs` uses is bypassing the check entirely. I've seen this happen before in some ubuntu machines, but not all... it's a very odd bug. @whyrusleeping does it happen to you in arch? it's never happened to me on osx.

> removing the files of course didn't help until i restarted the daemon

it should--
```
rm ~/.go-ipfs/logs/events-*
```
should work nicely.

**Comment by anarcat** [comment]
i am behind NAT, not sure about my friend, but at this point it doesn't matter: i'm just doing `ipfs add`, so my friend is not involved, right?

this is on Debian Jessie AMD64.

```
rm ~/.go-ipfs/logs/events-*
```

... "works" (ie. the files are gone), but since the daemon is still running and still holds the file descriptors open, the space isn't reclaimed.

**Comment by jbenet** [comment]
> but since the daemon is still running and still holds the file descriptors open, the space isn't reclaimed.

no, it should close the `events-<date>` files-- only hangs on to the latest (no date). It works on the machines i've tested-- is it not reclaiming for you? maybe you found a bug

**Comment by anarcat** [comment]
well, right. at least i think that's right: the `events-*` files are probably closed. but the `events.log` file is already huge, so that doesn't matter. :)

**Comment by whyrusleeping** [comment]
fixed in #1382
